src/Core/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -0,0 +1,354 @@
+                PlanName = plan.Name,
+                PlanType = plan.Type,
+                Seats = newSeatTotal,
+                PreviousSeats = organization.SmSeats
+            });
+
+        organization.SmSeats = (short?)newSeatTotal;
+
+        return paymentIntentClientSecret;
+    }
+
+    private async Task<string> AdjustServiceAccountsAsync(Organization organization, int serviceAccountAdjustment)
+    {
+        if (organization.SmServiceAccounts == null)
+        {
+            throw new BadRequestException("Organization has no Service Account limit, no need to adjust Service Account");
+        }
+
+        if (string.IsNullOrWhiteSpace(organization.GatewayCustomerId))
+        {
+            throw new BadRequestException("No payment method found.");
+        }
+
+        if (string.IsNullOrWhiteSpace(organization.GatewaySubscriptionId))
+        {
+            throw new BadRequestException("No subscription found.");
+        }
+
+        var plan = StaticStore.SecretManagerPlans.FirstOrDefault(p => p.Type == organization.PlanType);
+
+        if (plan == null)
+        {
+            throw new BadRequestException("Existing plan not found.");
+        }
+
+        if (!plan.HasAdditionalServiceAccountOption)
+        {
+            throw new BadRequestException("Plan does not allow additional Service Account.");
+        }
+
+        var newServiceAccountsTotal = organization.SmServiceAccounts.HasValue
+            ? organization.SmServiceAccounts.Value + serviceAccountAdjustment
+            : serviceAccountAdjustment;
+
+        if (plan.BaseServiceAccount > newServiceAccountsTotal)
+        {
+            throw new BadRequestException($"Plan has a minimum of {plan.BaseServiceAccount} Service Account.");
+        }
+
+        if (newServiceAccountsTotal <= 0)
+        {
+            throw new BadRequestException("You must have at least 1 Service Account.");
+        }
+
+        var additionalServiceAccounts = newServiceAccountsTotal - plan.BaseServiceAccount;
+
+        if (plan.MaxAdditionalServiceAccount.HasValue && additionalServiceAccounts > plan.MaxAdditionalServiceAccount.Value)
+        {
+            throw new BadRequestException($"Organization plan allows a maximum of " +
+                                          $"{plan.MaxAdditionalServiceAccount.Value} additional Service Account.");
+        }
+
+        if (!organization.SmServiceAccounts.HasValue || organization.SmServiceAccounts.Value > newServiceAccountsTotal)
+        {
+            var occupiedServiceAccounts = await _serviceAccountRepository.GetServiceAccountCountByOrganizationIdAsync(organization.Id);
+            if (occupiedServiceAccounts > newServiceAccountsTotal)
+            {
+                throw new BadRequestException($"Your organization currently has {occupiedServiceAccounts} seats filled. " +
+                                              $"Your new plan only has ({occupiedServiceAccounts}) seats. Remove some users.");
+            }
```

Comment:
Service accounts aren't really "occupied", suggested wording below.

````suggestion
            var currentServiceAccounts = await _serviceAccountRepository.GetServiceAccountCountByOrganizationIdAsync(organization.Id);
            if (currentServiceAccounts > newServiceAccountsTotal)
            {
                throw new BadRequestException($"Your organization currently has {currentServiceAccounts} Service Accounts. " +
                                              $"Your new plan only allows ({occupiedServiceAccounts}) Service Accounts. Remove some Service Accounts.");
            }
```

Reply:
Any idea why we talk about "your new plan" here? This flow is changing the number of service accounts, not the plan, as far as I know.

Reply:
sure